Firebase/Auth/Source/FIRAuthAppDelegateProxy.m

```diff
@@ -107,6 +107,53 @@ - (nullable instancetype)initWithApplication:(nullable UIApplication *)applicati
             didReceiveRemoteNotification:notification];
       }];
     }
+    SEL openURLOptionsSelector = @selector(application:openURL:options:);
+    SEL openURLAnnotationSelector = @selector(application:openURL:sourceApplication:annotation:);
+    SEL handleOpenURLSelector = @selector(application:handleOpenURL:);
+    if (&UIApplicationOpenURLOptionsAnnotationKey &&  // the constant is only available on iOS 9+
+        ([_appDelegate respondsToSelector:openURLOptionsSelector] ||
+         !([_appDelegate respondsToSelector:openURLAnnotationSelector] ||
+           [_appDelegate respondsToSelector:handleOpenURLSelector]))) {
```

Comment:
!(a || b) == !a && !b

Changed to the right side expression as it reads better to you.